assign-doi-datacite/src/main/java/no/unit/nva/doi/datacite/clients/MdsClient.java

```diff
@@ -0,0 +1,180 @@
+        }
+    }
+
+    private HttpRequest createDeleteMetadataRequest(CustomerConfig customer, Doi doi)
+        throws CustomerConfigException {
+        return HttpRequest.newBuilder()
+                   .DELETE()
+                   .uri(createUriForUpdatingMetadata(doi))
+                   .header(AUTHORIZATION_HEADER, customer.extractBasicAuthenticationString())
+                   .timeout(Duration.ofMillis(TIMEOUT))
+                   .build();
+    }
+
+    private void validateDeleteMetadataRequest(Doi doi) {
+        Objects.requireNonNull(doi, MISSING_DOI_IDENTIFIER_ARGUMENT);
+    }
+
+    private HttpRequest createDeleteDraftDoiRequest(CustomerConfig customer, Doi doi)
+        throws CustomerConfigException {
+        return HttpRequest.newBuilder()
+                   .DELETE()
+                   .header(AUTHORIZATION_HEADER, customer.extractBasicAuthenticationString())
+                   .uri(createUriForAccessingDoi(doi))
+                   .timeout(Duration.ofMillis(TIMEOUT))
+                   .build();
+    }
+
+    private void validateDeleteDraftDoiRequest(Doi doi) {
+        Objects.requireNonNull(doi, MISSING_DOI_IDENTIFIER_ARGUMENT);
+    }
+
+    private void validateUpdateMetadataInput(Doi doi, String metadataDataCiteXml) {
+        Objects.requireNonNull(doi, MISSING_DOI_IDENTIFIER_ARGUMENT);
+        Objects.requireNonNull(metadataDataCiteXml, MISSING_DATACITE_XML_ARGUMENT);
+    }
+
+    private HttpRequest createPostMetadataRequest(CustomerConfig customer,
+                                                  Doi doi,
+                                                  String metadataDataCiteXml)
+        throws CustomerConfigException {
+        return HttpRequest.newBuilder()
+                   .header(HttpHeaders.CONTENT_TYPE, APPLICATION_XML_CHARSET_UTF_8)
+                   .header(AUTHORIZATION_HEADER, customer.extractBasicAuthenticationString())
+                   .uri(createUriForUpdatingMetadata(doi))
+                   .timeout(Duration.ofMillis(TIMEOUT))
+                   .POST(HttpRequest.BodyPublishers.ofString(metadataDataCiteXml))
+                   .build();
+    }
+
+    private URI createUriForUpdatingMetadata(Doi doi) {
+        return UriWrapper.fromUri(dataciteMdsUri)
+                   .addChild(DATACITE_PATH_METADATA)
+                   .addChild(doi.toIdentifier())
+                   .getUri();
+    }
+
+    private HttpRequest createLandingPagePutRequest(CustomerConfig customer,
+                                                    Doi doi,
+                                                    URI landingPage)
+        throws CustomerConfigException {
+        return HttpRequest.newBuilder()
+                   .header(HttpHeaders.CONTENT_TYPE, TEXT_PLAIN_CHARSET_UTF_8)
+                   .header(AUTHORIZATION_HEADER, customer.extractBasicAuthenticationString())
+                   .timeout(Duration.ofMillis(TIMEOUT))
+                   .uri(createUriForAccessingDoi(doi))
```

Comment:
`createUriForAccessingDoi ` is the same as is used in `deleteDraftDoi`, but maybe that is meant to be?